cli/src/lib/npm/npmLibDefs.js

```diff
@@ -244,13 +244,22 @@ function validateVersionNumPart(part: string, partName: string): number {
 }
 
 function pkgVersionMatch(pkgSemver: string, libDefSemverRaw: string) {
-  // The package version should be treated as a semver implicitly prefixed by a
-  // `^`. (i.e.: "foo_v2.2.x" is the same range as "^2.2.x")
+  // The package version should be treated as a semver implicitly prefixed by
+  // `^` or `~`. Depending on whether or not the minor value is defined.
+  // i.e.: "foo_v2.2.x" is the same range as "~2.2.x"
+  //        and "foo_v2.x.x" is the same range as "^2.x.x"
   // UNLESS it is prefixed by the equals character (i.e. "foo_=v2.2.x")
-  let libDefSemver =
-    libDefSemverRaw[0] !== '=' && libDefSemverRaw[0] !== '^'
-      ? '^' + libDefSemverRaw
-      : libDefSemverRaw;
+  const libDefSemver = (() => {
+    const versionSplit = libDefSemverRaw.split('.');
+    if (libDefSemverRaw[0] !== '=' && libDefSemverRaw[0] !== '^') {
+      if (versionSplit[1] !== 'x') {
```

Comment:
This should handle if versions are `0.4.x` for beta library versions

Reply:
actually should be handled fine https://docs.npmjs.com/cli/v6/using-npm/semver#tilde-ranges-123-12-1